airflow/dag_processing/parse_info.py

```diff
@@ -0,0 +1,98 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from functools import cached_property
+from pathlib import Path
+from typing import Any
+
+
+@dataclass(frozen=True)
+class ParseBundleInfo:
+    """
+    In-parsing time context about bundle being processed.
+
+    :param name: Bundle name.
+    :param path: Path of the bundle version's root.
+    :param version: Bundle version.
+    """
+
+    name: str
+    path: Path | str = field(compare=False)
+    version: str | None = None
+
+
+@dataclass(frozen=True)
+class _ParseFileInfo:
+    rel_path: Path | str
+    bundle: ParseBundleInfo
+
+
+@dataclass(frozen=True)
+class ParseFileInfo(_ParseFileInfo):
+    """
+    Information about a DAG file at parse time with fixed version.
```

Comment:
when you say "with fixed version" .... it doesn't seem like you really guarantee that since version is optional in ParseBundleInfo

Reply:
It was referred to a "version" as "whatever presented as a version of a bundle data", not as "version uid". Changed to "resolved bundle metadata".